### PR TITLE
listen to http when http2 or https used

### PR DESCRIPTION
**Pull request by Shuunen**
@ericmdantas : this is still a WIP but it works :)

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -353,6 +353,15 @@ module.exports = class Server extends EventEmitter {
 
   _createServer() {
     this._httpServer = this._createHttpServer();
+
+    if (this.opts.http2 || this.opts.secure) {
```
Comment (ericmdantas) [inline lib/server.js]: You can check only `this.opts.secure`, the `http2` options forces that.

File: lib/server.js [hunk 2]
```diff
@@ -353,6 +353,15 @@ module.exports = class Server extends EventEmitter {
...
+      let port = 80 // TODO : this should be this.opts.httpPort or this.opts.insecurePort & 80 by default if none provided
```
Comment (ericmdantas) [inline lib/server.js]: Note to self: this comment addresses good points, those options should exist.

**Comment by ericmdantas** [comment]
Thanks for the PR, @Shuunen! It's looking great already.

I left two minor comments, one is a reminder to myself and the other one is a tip to use an option that's already normalized. Other than that, it's good to go.

File: lib/server.js (hunk 2 above)
Comment (Shuunen) [reply lib/server.js]: @ericmdantas what should we use here `httpPort` , `insecurePort` , or something else ?

File: lib/server.js (hunk 1 above)
Comment (Shuunen) [reply lib/server.js]: fixed :+1:

**Comment by Shuunen** [comment]
so I added all I think was required with comments and options to let user customize everything :)

let me know if you like :)

**Comment by Shuunen** [comment]
how to sastisfy coverage/coveralls ?

**Comment by ericmdantas** [comment]
Awesome, thanks!

Don't worry about coveralls, the creation of a server is too annoying to test, so you can leave it as it is. Let me know when it's ready to be merged.

File: lib/server.js [hunk 3]
```diff
@@ -106,6 +106,14 @@ module.exports = class Server extends EventEmitter {
         long: 'port'
       },
       {
+        short: 'ip',
```
Comment (ericmdantas) [inline lib/server.js]: The short version could cause a bit of confusion... maybe use `insPort`?

File: lib/server.js [hunk 4]
```diff
@@ -106,6 +106,14 @@ module.exports = class Server extends EventEmitter {
...
+        long: 'insecurePort'
+      },
+      {
+        short: 'rh',
```
Comment (ericmdantas) [inline lib/server.js]: Maybe use `rhh`?

File: lib/server.js (hunk 3 above)
Comment (Shuunen) [reply lib/server.js]: sure

File: lib/server.js (hunk 4 above)
Comment (Shuunen) [reply lib/server.js]: yup

**Comment by Shuunen** [comment]
done :)

**Comment by ericmdantas** [comment]
Thanks!

Is it ready to be merged? If so, I'll add some tests and update the readme.

**Comment by Shuunen** [comment]
yes it's good to me :+1:

**Comment by ericmdantas** [comment]
Alright, thanks once again!

**Comment by ericmdantas** [comment]
Merged. I'll add some tests and docs and I'll release a minor version in a moment.

**Pull request merged**